test/test_hmc_util.py

```diff
@@ -297,3 +300,70 @@ def test_warmup_adapter(jitted):
     # Verifies that inverse_mass_matrix does not change during the last window
     # despite z_flat changes w.r.t time t,
     assert_allclose(final_inverse_mass_matrix, inverse_mass_matrix)
+
+
+@pytest.mark.parametrize('leaf_idx, ckpt_idxs', [
+    (6, (3, 2)),
+    (7, (0, 2)),
+    (13, (2, 2)),
+    (15, (0, 3)),
+])
+def test_leaf_idx_to_ckpt_idx(leaf_idx, ckpt_idxs):
+    assert _leaf_idx_to_ckpt_idxs(leaf_idx) == ckpt_idxs
+
+
+@pytest.mark.parametrize('ckpt_idxs, expected_turning', [
+    ((3, 2), False),
+    ((3, 3), True),
+    ((0, 0), False),
+    ((0, 1), True),
+    ((1, 3), True),
+])
+def test_is_iterative_turning(ckpt_idxs, expected_turning):
+    inverse_mass_matrix = np.ones(1)
+    r = 1.
+    r_sum = 3.
+    r_ckpts = np.array([1., 2., 3., -2.])
+    r_sum_ckpts = r_ckpts + 1
+
+    actual_turning = _is_iterative_turning(inverse_mass_matrix, r, r_sum, r_ckpts, r_sum_ckpts,
+                                           *ckpt_idxs)
+    assert expected_turning == actual_turning
+
+
+@pytest.mark.parametrize('step_size', [0.01, 1., 100.])
+@pytest.mark.parametrize('iterative_build', [True, False])
+def test_build_tree(step_size, iterative_build):
```

Comment:
I think it will be great to have a test that `iterative_build = True, False` return the same tree for different models (I think with the same rng they would), which can be deleted once we make this the default. But in the meantime, it will be nice to use this code path for testing.